api/v1alpha1/valkeycluster_types.go

```diff
@@ -73,36 +90,81 @@ type ExporterSpec struct {
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// For Kubernetes API conventions, see:
-	// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties
-
-	// conditions represent the current state of the ValkeyCluster resource.
-	// Each condition has a unique type and reflects the status of a specific aspect of the resource.
-	//
-	// Standard condition types include:
-	// - "Available": the resource is fully functional
-	// - "Progressing": the resource is being created or updated
-	// - "Degraded": the resource failed to reach or maintain its desired state
-	//
-	// The status of each condition is one of True, False, or Unknown.
+	// State provides a high-level summary of the cluster's current state.
+	// +kubebuilder:default=Initializing
+	// +optional
+	State ClusterState `json:"state,omitempty"`
+
+	// Reason provides a brief machine-readable explanation for the current state.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// Message provides human-readable details about the current state.
+	// +optional
+	Message string `json:"message,omitempty"`
+
+	// Shards represents the number of shards currently formed in the cluster.
+	// +kubebuilder:default=0
+	// +optional
+	Shards int32 `json:"shards,omitempty"`
+
+	// ReadyShards represents the number of shards that are fully healthy.
+	// +kubebuilder:default=0
+	// +optional
+	ReadyShards int32 `json:"readyShards,omitempty"`
+
+	// Conditions represent the current state of the ValkeyCluster resource.
+	// Standard condition types:
+	// - "Ready": the cluster is fully functional and serving traffic
+	// - "Progressing": the cluster is being created, updated, or scaled
+	// - "Degraded": the cluster is impaired but may be partially functional
+	// Valkey-specific condition types:
+	// - "ClusterFormed": all nodes have joined and meet the shard/replica layout
+	// - "SlotsAssigned": all 16384 hash slots are assigned to primaries
 	// +listType=map
 	// +listMapKey=type
 	// +optional
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
+const (
+	ConditionReady         = "Ready"
+	ConditionProgressing   = "Progressing"
+	ConditionDegraded      = "Degraded"
+	ConditionClusterFormed = "ClusterFormed"
+	ConditionSlotsAssigned = "SlotsAssigned"
+)
+
+const (
+	// Common reasons for conditions
+	ReasonInitializing      = "Initializing"
+	ReasonReconciling       = "Reconciling"
+	ReasonClusterHealthy    = "ClusterHealthy"
+	ReasonServiceError      = "ServiceError"
+	ReasonConfigMapError    = "ConfigMapError"
+	ReasonDeploymentError   = "DeploymentError"
+	ReasonPodListError      = "PodListError"
+	ReasonAddingNodes       = "AddingNodes"
+	ReasonNodeAddFailed     = "NodeAddFailed"
+	ReasonMissingShards     = "MissingShards"
+	ReasonMissingReplicas   = "MissingReplicas"
+	ReasonReconcileComplete = "ReconcileComplete"
+	ReasonClusterFormed     = "TopologyComplete"
```

Comment:
This was changed in the last fixux. I think it's good if the reason type name matches the string output.
```suggestion
	ReasonTopologyComplete     = "TopologyComplete"
```